File: Python/409. Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        if s is None or s == "":
            return 0

        count = 0
        check = set()
        for i in range(len(s)):
            if s[i] in check:
                check.remove(s[i])
                count += 1
            else:
                check.add(s[i])

        if len(check) != 0:
            return count * 2 + 1
        return count * 2

File: PHP/409. Longest Palindrome.php
<?php

/**
 *  @noinspection PhpIllegalPsrClassPathInspection
 *  @noinspection PhpMultipleClassDeclarationsInspection
 */
class Solution
{
    public function longestPalindrome(string $s): int
    {
        if (empty($s) || empty(strlen($s))) {
            return 0;
        }

        $count = 0;
        $check = [];
        for ($i = 0; $i < strlen($s); $i++) {
            if (isset($check[$s[$i]])) {
                unset($check[$s[$i]]);
                $count++;
            } else {
                $check[$s[$i]] = 1;
            }
        }
        if (count($check) != 0) {
            return $count * 2 + 1;
        }

        return $count * 2;
    }
}
